pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java

```diff
@@ -702,6 +740,28 @@ public void copyToLocalFile(URI srcUri, File dstFile)
     _s3Client.getObject(getObjectRequest, ResponseTransformer.toFile(dstFile));
   }
 
+  @Override
+  public void copyToLocalFile(URI srcUri, File dstFile)
+      throws Exception {
+    try {
+      LOGGER.info("Copy {} to local {}", srcUri, dstFile.getAbsolutePath());
+      copyToLocalFileInternal(srcUri, dstFile);
+    } catch (Exception e) {
+      LOGGER.warn("Caught exception during S3 copy, attempting to refresh credentials and retry");
```

Comment:
I feel we also need to handle the refresh logic in other method? E.g. mkdir, list files, delete file etc.